Surface_mesh_topology/include/CGAL/Polygonal_schema.h

```diff
@@ -14,10 +14,11 @@
 
 #include <CGAL/license/Surface_mesh_topology.h>
 
-#ifdef _MSC_VER
+// TODO ADD SOME TEST IF WE ARE ON WINDOWS ?
+// THIS SEEMS A WRONG TEST =>  #ifdef _MSC_VER
 // To remove the warning on windows about unsafe strncpy function.
 #define _CRT_SECURE_NO_WARNINGS
-#endif
+// #endif
```

Comment:
[Code smell]!

1. That is very nasty, for our users, to deactivate a security feature of MSVC silently.
2. Why don't you use `strncpy_s`?
3. Actually, why are you using raw C strings in the first place? In C++, we have `std::string` to avoid a lot of nasty bugs.

[Code smell]: https://en.wikipedia.org/wiki/Code_smell

Reply:
1. My goal is to remove this warning only in my file. But for now it does not remove the warning, this is why I did some tests.
2. I was not able to use strncpy_s on my os; even with "__STDC_WANT_LIB_EXT1__ ". 
3. For memory reason; these labels are constants thus I don't need all the possibilities of std::string.

Reply:
I also personally think that we should reject any new code in CGAL that uses `new`/`delete`.

You should definitely use `std::string` instead of `const char*` and `new char[s.size()+1]` everywhere. At least with `std::string` you will benefit from improvements like the "small buffer optimization". The downside is that `sizeof(std::string)` is bigger that `sizeof(char*)`. But you might recover that memory with the small buffer optimization, if your labels are short enough.

Reply:
Ok will do...

Reply:
If your labels are constants, then do not copy them with `strncopy`, and just hold the pointer to the constant strings.

Do you really gain something by not using `std::string`? Even if you do not "need all the possibilities". That is not a good reason: I have never needed all the possibilities of `std::vector`, but I will never use a C-array instead.

Reply:
Still not convinced. One ptr: 8 bytes; One std::string between 24 and 40 bytes. For 500000 elements, I waste between 7 Mb and 15 Mb of memory for nothing...

Reply:
If you use-case is constant strings anyway, then only hold the pointers to constant strings, instead of copying the string in all nodes.

Reply:
They are not constant strings; they are constant, i.e. never updated during their life.

Reply:
Is there any reason the node should hold a copy of the string, instead of referencing existing copies? If the answer is yes, then I would say "use `std::string`. If the answer is no, then you should use non-owning pointers to raw C-strings, without the `new/strncopy/delete` machinery.